htdocs/install/mysql/migration/repair.sql

```diff
@@ -416,10 +407,33 @@ drop table tmp_bank_url_expense_user;
 
 
 -- Delete duplicate accounting account, but only if not used
-DROP TABLE tmp_llx_accouting_account;
-CREATE TABLE tmp_llx_accouting_account AS SELECT MIN(rowid) as MINID, account_number, entity, fk_pcg_version, count(*) AS NB FROM llx_accounting_account group BY account_number, entity, fk_pcg_version HAVING count(*) >= 2 order by account_number, entity, fk_pcg_version;
---SELECT * from tmp_llx_accouting_account;
-DELETE from llx_accounting_account where rowid in (select minid from tmp_llx_accouting_account where minid NOT IN (SELECT fk_code_ventilation from llx_facturedet) AND minid NOT IN (SELECT fk_code_ventilation from llx_facture_fourn_det) AND minid NOT IN (SELECT fk_code_ventilation from llx_expensereport_det));
+DROP TABLE IF EXISTS tmp_llx_accouting_account;
+CREATE TABLE tmp_llx_accouting_account AS
+SELECT
+    aa.rowid,
+    aad.min
+FROM llx_accounting_account AS aa
+         INNER JOIN (
+    SELECT
+        account_number,
+        entity,
+        fk_pcg_version,
+        MIN(rowid) AS min
+    FROM llx_accounting_account
+    GROUP BY account_number, entity, fk_pcg_version
+    HAVING COUNT(*) >= 2
+) AS aad ON
+            aa.account_number = aad.account_number
+        AND aa.entity = aad.entity
+        AND aa.fk_pcg_version = aad.fk_pcg_version
+        AND aa.rowid != aad.min;
+-- SELECT * from tmp_llx_accouting_account;
+-- If there is record in tmp_llx_accouting_account, make a look on each line to do
+UPDATE llx_facturedet        AS t LEFT JOIN tmp_llx_accouting_account AS taa ON taa.rowid = t.fk_code_ventilation SET t.fk_code_ventilation = taa.min WHERE taa.rowid IS NOT NULL;
```

Comment:
I am not sure this kind of syntax Update xxx left join yyy is standard sql accepted by pgsql. Did you test it in postgres ?

Reply:
it's in /mysql/migration/repair.sql so is it not executed only for mysql ?

Reply:
No, it is common to all databases.
Mysql is the database to use for main sql code (so contains allnpart common)  and other dir for specific. Repair is currently common so found into the mysql dir.

Reply:
for me https://github.com/Dolibarr/dolibarr/commit/d9d0618eec42f549e82c21d3178f8221a8077550 seems to be ok to fix that bug because entity is added into group by but did somebody have a sql dump with the bug and could you please send me by private mail to run and tests the repair process with pgsql and mariadb/mysql if you can't do it by yourself @kkhelifa-opendsi ?